cli/npm/resolvers/common.rs

```diff
@@ -95,24 +97,47 @@ pub fn ensure_registry_read_permission(
   permissions: &dyn NodePermissions,
   registry_path: &Path,
   path: &Path,
+  registry_cache: &Mutex<HashMap<PathBuf, PathBuf>>,
+  path_cache: &Mutex<HashMap<PathBuf, PathBuf>>,
```

Comment:
Since this function essentially has state now, it could be turned into a struct and these details buried within it. For example:

```rs
pub struct RegistryReadPermissionChecker {
  fs: Arc<dyn FileSystem>,
  cache: Mutex<HashMap<PathBuf, PathBuf>>, // I think we only need a single cache?
  registry_path: PathBuf,
}

impl RegistryReadPermissionChecker {
  pub fn new(fs: Arc<dyn FileSystem>, registry_path: PathBuf) -> Self {
    Self {
      fs,
      registry_path,
      cache: Default::default(),
    }
  }
  
  pub fn ensure_registry_read_permission(
    permissions: &dyn NodePermissions,
    path: &Path,
  } -> Result<(), AnyError> {
    // etc...
  }
}
```

Then we can create and use this struct within both global.rs and local.rs.